src/sentry/issues/group_policy.py

```diff
@@ -0,0 +1,167 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Dict, Optional, Set, Type
+
+from django.conf import settings
+
+from sentry.issues.grouptype import GroupType
+from sentry.models import Organization, Project
+from sentry.utils import metrics, redis
+
+if TYPE_CHECKING:
+    from sentry.utils.performance_issues.performance_detection import PerformanceProblem
+
+_group_policy_registry: Dict[int, Type[GroupPolicy]] = {}
+DEFAULT_IGNORE_LIMIT: int = 3
+DEFAULT_EXPIRY_TIME: int = 60 * 60 * 24
+
+
+@dataclass(frozen=True)
+class NoiseConfig:
+    ignore_limit: int = DEFAULT_IGNORE_LIMIT
+    expiry_time: int = DEFAULT_EXPIRY_TIME
+
+
+@dataclass(frozen=True)
+class GroupPolicy:
+    group_type_id: int
```

Comment:
This should probably be validated against existing `GroupType` `type_id` values, idk if there's an easy way to build a list of choices or maybe just another check in `__init_subclass__` could suffice.